agent/stats/common_test.go

```diff
@@ -170,6 +179,63 @@ func validateMetricsMetadata(metadata *ecstcs.MetricsMetadata) error {
 	return nil
 }
 
+func validateHealthMetricsMetadata(metadata *ecstcs.HealthMetadata) error {
+	if metadata == nil {
+		return fmt.Errorf("metadata is nil")
+	}
+
+	if aws.StringValue(metadata.Cluster) != defaultCluster {
+		return fmt.Errorf("expected cluster in metadata to be: %s, got %s",
+			defaultCluster, aws.StringValue(metadata.Cluster))
+	}
+
+	if aws.StringValue(metadata.ContainerInstance) != defaultContainerInstance {
+		return fmt.Errorf("expected container instance in metadata to be %s, got %s",
+			defaultContainerInstance, aws.StringValue(metadata.ContainerInstance))
+	}
+	if len(aws.StringValue(metadata.MessageId)) == 0 {
+		return fmt.Errorf("empty MessageId")
+	}
+
+	return nil
+}
+
+func validateContainerHealthMetrics(metrics []*ecstcs.ContainerHealth, expected int) error {
+	if len(metrics) != expected {
+		return fmt.Errorf("mismatch in number of ContainerHealth elements. Expected: %d, Got: %d",
+			expected, len(metrics))
+	}
+	for _, health := range metrics {
+		if aws.StringValue(health.ContainerName) == "" {
+			return fmt.Errorf("container name is empty")
+		}
+		if aws.StringValue(health.HealthStatus) == "" {
+			return fmt.Errorf("container health status is empty")
+		}
+		if aws.TimeValue(health.StatusSince).IsZero() {
+			return fmt.Errorf("container health status change timestamp is empty")
+		}
+	}
+	return nil
+}
+
+func validateTaskHealthMetrics(t *testing.T, engine *DockerStatsEngine) {
+	healthMetadata, healthMetrics, err := engine.GetTaskHealthMetrics()
+	assert.NoError(t, err, "getting task health metrics failed")
+	require.Len(t, healthMetrics, 1)
+	assert.NoError(t, validateHealthMetricsMetadata(healthMetadata), "validating health metedata failed")
+	assert.Equal(t, aws.StringValue(healthMetrics[0].TaskArn), taskArn, "task arn not expected")
+	assert.Equal(t, aws.StringValue(healthMetrics[0].TaskDefinitionFamily), taskDefinitionFamily, "task definition family not expected")
+	assert.Equal(t, aws.StringValue(healthMetrics[0].TaskDefinitionVersion), taskDefinitionVersion, "task definition version not expected")
+	assert.NoError(t, validateContainerHealthMetrics(healthMetrics[0].Containers, 1))
+}
+
+func validateEmptyTaskHealthMetrics(t *testing.T, engine *DockerStatsEngine) {
+	_, healthMetrics, err := engine.GetTaskHealthMetrics()
```

Comment:
can you also validate the metadata here? Or, please document why you aren't validating it.

Reply:
It shouldn't matter, because if the metrics is empty, then we don't care the metadata as we won't send health metrics in this case.

Reply:
thanks for that explanation. Can you please document the same in the method?

Reply:
done